Rules/AvoidLongLines.cs

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+#if !CORECLR
+using System.ComponentModel.Composition;
+#endif
+using System.Globalization;
+using System.Management.Automation.Language;
+using Microsoft.Windows.PowerShell.ScriptAnalyzer.Generic;
+
+namespace Microsoft.Windows.PowerShell.ScriptAnalyzer.BuiltinRules
+{
+    /// <summary>
+    /// AvoidLongLines: Checks for lines longer than 120 characters
+    /// </summary>
+#if !CORECLR
+    [Export(typeof(IScriptRule))]
+#endif
+    public class AvoidLongLines : ConfigurableRule
+    {
+        /// <summary>
+        /// Construct an object of AvoidLongLines type.
+        /// </summary>
+        public AvoidLongLines() : base()
+        {
+            // Enable the rule by default
+            Enable = false;
+        }
+
+        [ConfigurableRuleProperty(defaultValue: 120)]
+        public int LineLength { get; set; }
+        /// <summary>
+        /// Analyzes the given ast to find violations.
+        /// </summary>
+        /// <param name="ast">AST to be analyzed. This should be non-null</param>
+        /// <param name="fileName">Name of file that corresponds to the input AST.</param>
+        /// <returns>A an enumerable type containing the violations</returns>
+        public override IEnumerable<DiagnosticRecord> AnalyzeScript(Ast ast, string fileName)
+        {
+            if (ast == null)
+            {
+                throw new ArgumentNullException("ast");
+            }
+
+            var diagnosticRecords = new List<DiagnosticRecord>();
+
+            string[] lines = Regex.Split(ast.Extent.Text, @"\r?\n");
```

Comment:
I've made this change, but @rjmholt if you don't mind, why is this the preferable choice over the regex? In my head the one-line regex solution was more tidy than your suggestion which tells me there's a part of your suggestion that I don't fully understand.

Reply:
Basically regex is expensive. There are various ways to make a regex work internally and the *cheapest* (amortising over many operations) is to allocate a data structure representing a state machine. Under the hood, I believe that's what .NET will do with a static call like this one after enough calls.

Before then though, it will use a regex interpreter. This still involves a state machine of sorts, but is more involved.

Basically a regex is cheap enough to write, but quite costly to run, *compared to* searching for a substring. There are excellent algorithms for substring search (why `grep` is so fast). So using this line separators method looks a bit uglier on the outside, but the actual code executed as a result is likely to be simpler. *Even if we expect the regex to optimise based on it being given literal patterns*.